easybuild/easyconfigs/i/intel-compilers/intel-compilers-2023.2.1.eb

```diff
@@ -0,0 +1,37 @@
+name = 'intel-compilers'
+version = '2023.2.1'
+
+homepage = 'https://software.intel.com/content/www/us/en/develop/tools/oneapi/hpc-toolkit.html'
+description = "Intel C, C++ & Fortran compilers (classic and oneAPI)"
+
+toolchain = SYSTEM
+
+# see https://software.intel.com/content/www/us/en/develop/articles/oneapi-standalone-components.html
+sources = [
+    {
+        'source_urls': [
+            'https://registrationcenter-download.intel.com/akdlm/IRC_NAS/ebf5d9aa-17a7-46a4-b5df-ace004227c0e/',
+        ],
+        'filename': 'l_dpcpp-cpp-compiler_p_%(version)s.8_offline.sh',
+    },
+    {
+        'source_urls': [
+            'https://registrationcenter-download.intel.com/akdlm/IRC_NAS/0d65c8d4-f245-4756-80c4-6712b43cf835/',
+        ],
+        'filename': 'l_fortran-compiler_p_%(version)s.8_offline.sh',
+    },
+]
+checksums = [
+    {'l_dpcpp-cpp-compiler_p_2023.2.1.8_offline.sh':
+     'f5656b2f5bb5d904639e6ef1f90a2d2e760d2906e82ebc0dd387709738ca714b'},
+    {'l_fortran-compiler_p_2023.2.1.8_offline.sh':
+     'd4e36abc014c184698fec318a127f15a696b5333b3b0282aba1968b351207185'},
+]
+
+local_gccver = '12.3.0'
```

Comment:
Not sure if we should go with 12.3.0 or 13.2.0.

Reply:
I found the following in https://www.intel.com/content/www/us/en/developer/articles/release-notes/oneapi-c-compiler-release-notes.html:

> Note: Intel® C++ Compiler Classic doesn't support GCC 13 compatibility mode. For GCC 13 compatibility mode, please transition to Intel®  oneAPI DPC++/C++ Compiler.

Reply:
Done in https://github.com/easybuilders/easybuild-easyconfigs/pull/18439/commits/ccaeebc51b3d523d55a206aab7449c5b3380f410